editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/FuseIntegrationProjectWizardTemplatePage.java

```diff
@@ -253,9 +254,42 @@ private void setTemplatesActive(boolean active) {
 			btn_blueprintDSL.setEnabled(getSelectedTemplate() != null && getSelectedTemplate().getTemplate().supportsDSL(CamelDSLType.BLUEPRINT));
 			btn_springDSL.setEnabled(getSelectedTemplate() != null && getSelectedTemplate().getTemplate().supportsDSL(CamelDSLType.SPRING));
 			btn_javaDSL.setEnabled(getSelectedTemplate() != null && getSelectedTemplate().getTemplate().supportsDSL(CamelDSLType.JAVA));
+			updateDSLButtonGroup(getSelectedTemplate());
+		}
+	}
+	
+	private void updateDSLButtonGroup(TemplateItem template) {
+		if (template == null || !disabledDSLSelected()) return;
+		for (CamelDSLType dslType : CamelDSLType.values()) {
+			if (template.getTemplate().supportsDSL(dslType)) {
+				selectButtonForDSL(dslType);
+				return;
+			}
 		}
 	}
 	
+	private void selectButtonForDSL(CamelDSLType dsltype) {
+		if (dsltype.equals(CamelDSLType.BLUEPRINT)) {
+			btn_blueprintDSL.setSelection(true);
+			btn_javaDSL.setSelection(false);
+			btn_springDSL.setSelection(false);
+		} else if (dsltype.equals(CamelDSLType.JAVA)) {
+			btn_blueprintDSL.setSelection(false);
+			btn_javaDSL.setSelection(true);
+			btn_springDSL.setSelection(false);
+		} else {
+			btn_blueprintDSL.setSelection(false);
+			btn_javaDSL.setSelection(false);
+			btn_springDSL.setSelection(true);
+		}
+	}
+	
+	private boolean disabledDSLSelected() {
+		return 	(btn_blueprintDSL.getSelection() == true && btn_blueprintDSL.isEnabled() == false) ||
+				(btn_javaDSL.getSelection() == true && btn_javaDSL.isEnabled() == false) ||
```

Comment:
can extract a method having a button in parameter, somethign like
private boolean isSelectedAndDisabled(Button button){
return button.getSelection() && !button.isEnabled();
}